lib/net/ssh/transport/algorithms.rb

```diff
@@ -290,10 +290,26 @@ def compose_algorithm_list(supported, option, append_all_supported_algorithms =
           list = []
           option = Array(option).compact.uniq
 
-          if option.first && option.first.start_with?('+')
+          if option.first && option.first.start_with?('+', '-')
             list = supported.dup
-            list << option.first[1..-1]
-            list.concat(option[1..-1])
+
+            appends = option.select { |opt| opt.start_with?('+') }.map { |opt| opt[1..-1] }
+            deletions = option.select { |opt| opt.start_with?('-') }.map { |opt| opt[1..-1] }
+
+            list.concat(appends) if appends.any?
+
+            if deletions.any?
```

Comment:
remove any? you can iterate on a empty array

Reply:
ah yeah, good call. done in c457c06